### ISSUE TITLE
Type safety issue

### ISSUE THREAD

**Issue by hyperreal-gopher** [hyperreal-gopher/waffle#102]
```
waffle: Maybe.fromJust: Nothing
CallStack (from HasCallStack):
  error, called at libraries/base/Data/Maybe.hs:148:21 in base:Data.Maybe
  fromJust, called at src/UI/Progress.hs:151:60 in waffle-0.4.0.0-inplace-waffle-library:UI.Progress
```

Happens while trying to goto gopher.floodgap.org

**Comment by hyperreal-gopher** [comment]
It's worth noting the intended address was `gopher.floodgap.com` (.com)

**Comment by hyperreal-gopher** [comment]
I believe this was solved in v0.8.0
